### PR TITLE
add restore trash can

### PR DESCRIPTION
**Pull request by itkg-alavieille**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1035
https://github.com/open-orchestra/open-orchestra-model-interface/pull/110

### CODE REVIEW

File: BackofficeBundle/Resources/public/coffee/table/TableviewRestoreAction.coffee [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      yesCallback: (params) ->
+        $.ajax
+          url: params.url
+          method: 'PUT'
+          success : () ->
+            params.row.hide()
+            options.table.fnSettings().clearCache = true
+            Backbone.history.loadUrl(Backbone.history.fragment)
+          error: (jqXHR) ->
+            viewClass = appConfigurationView.getConfiguration('status', 'apiError')
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/table/TableviewRestoreAction.coffee]: For this class, the dom container is useless (`DisplayApiErrorView`)

File: Backoffice/RestoreEntity/Strategies/RestoreNodeStrategy.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        return $entity instanceof NodeInterface;
+    }
+
+    /**
+     * @param NodeInterface $node
+     */
+    public function restore($node)
+    {
+        $this->nodeManager->restoreNode($node);
```
Comment (nicolasThal) [inline Backoffice/RestoreEntity/Strategies/RestoreNodeStrategy.php]: same questino

File: Backoffice/RestoreEntity/Strategies/RestoreContentStrategy.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        return $entity instanceof ContentInterface;
+    }
+
+    /**
+     * @param ContentInterface $content
+     */
+    public function restore($content)
+    {
+        $this->contentManager->restoreContent($content);
```
Comment (itkg-alavieille) [inline Backoffice/RestoreEntity/Strategies/RestoreContentStrategy.php]: Yes but where do you put them, I can put in ContentStrategy or NodeStrategy but in this case, they will use components of Symfony so they will not be in the business logic layer

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1: with green tests :)

**Pull request merged**